### PR TITLE
Feature(analyzer): fix type casting with conditional types

### PR DESCRIPTION
**Pull request by sunrabbit123**
**BREAKING CHANGE:**
```ts
type ZeroOf<T extends number | string | boolean> = T extends number ? 0 : T extends string ? "" : false;

function zeroOf<T extends number | string | boolean>(value: T) {
    return <ZeroOf<T>>(typeof value === "number" ? 0 : typeof value === "string" ? "" : false);
}

export function f21<T extends number | string>(x: T, y: ZeroOf<T>) {
    let z1: number | string = y;
    let z2: 0 | "" = y;
}
```

### CODE REVIEW

File: crates/stc_ts_file_analyzer/src/analyzer/expr/type_cast.rs [hunk 1]
```diff
@@ -106,7 +106,7 @@ impl Analyzer<'_, '_> {
         casted_ty.freeze();
 
         self.validate_type_cast_inner(span, &orig_ty, &casted_ty).report(&mut self.storage);
-
+        dbg!(&casted_ty);
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/type_cast.rs:109]: Please remove this

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
Thanks!

**Comment by sunrabbit123** [comment]
this resolve #922 #921

**Comment by kdy1** [comment]
Did you try `./auto-unignore.sh`? If CI is not broken, CI should fail if it's really resolved

**Comment by sunrabbit123** [comment]
this duplicate pr #964
The work overlaps, is it okay?

It looks like the difference between front-end and back-end processing.
And the code on the #964 looks cleaner and has less overhead

**Comment by kdy1** [comment]
Ah, I'll revert this then

**Comment by kdy1** [comment]
Wait, do you think it's fine to revert this PR? Do two PRs semantically identical?

**Comment by sunrabbit123** [comment]
#964 recursively processes a conditional value.
#968 recursively pulls out the true and false values from the front of the conditional.

The difference is that #968 extracts the union type and uses the
#964 recursively checks them one by one.

I think they mean the same thing, but

I'm curious to know what your opinion is.

**Pull request merged**